inc/Integrations/Google/Sheets/GoogleSheetsIntegration.php

```diff
@@ -186,4 +187,28 @@ public static function get_block_registration_snippets( array $data_source_confi
 
 		return $snippets;
 	}
+
+	/**
+	 * Due to the fact that we are using the same query for both the get and list queries, and
+	 * only filtering out the results based on the input variables, the in-memory cache will not
+	 * work as expected. This enhances the request details to include the input variables, so that
+	 * the in-memory cache will be able to differenciate each request.
+	 *
+	 * @param array<string, mixed> $request_details The request details.
+	 * @param string $query The query being executed.
+	 * @param array<string, mixed> $input_variables The input variables for the current request.
+	 * @return array<string, array{
+	 *   method: string,
+	 *   options: array<string, mixed>,
+	 *   origin: string,
+	 *   uri: string,
+	 * }>
+	 */
+	public static function enhance_request_details( array $request_details, string $query, array $input_variables ): array {
```

Comment:
I've taken this approach on purpose. I didn't want to alter the request_details method as it's technically working as expected in this case. The actual request doesn't really have any enhancements and we really shouldn't have data source specific logic in there. 

Doing it this way means we let the integration handle it. Happy to have a discussion on this.